samples/tests/second/mod.rs

```diff
@@ -0,0 +1,30 @@
+samples::test! {
+    forward_of_reverse;
+    // ANCHOR: forward_of_reverse
+    #[autodiff(df, Reverse, Duplicated, Active)]
+    fn f(x: &[f32; 2]) -> f32 {
+        x[0] * x[0] + x[1] * x[0]
+    }
+
+    #[autodiff(dg, Forward, Dual, Dual)]
+    fn g(x: &[f32; 2], gx: &mut [f32; 2]) {
+        let mut bx = [0.0, 0.0];
+        df(x, &mut bx, 1.0);
+        *gx = bx;
+    }
+
+    fn main() {
+        let x = [2.0, 3.0];
+        let fx = f(&x);
+        assert_eq!(10.0, fx);
+
+        let mut gx = [0.0, 0.0];
+        g(&x, &mut gx);
+        assert_eq!([7.0, 2.0], gx);
+
+        let dx = [1.0, 0.0];
+        let mut dgx = [0.0, 0.0];
+        dg(&x, &dx, &mut gx, &mut dgx);
```

Comment:
This runs cleanly without this final block, which fails in the same way as the neohookean example.
```
[compiler/rustc_codegen_llvm/src/llvm/ffi.rs:879:5] &fnc = (ptr:; Function Attrs: noinline nonlazybind sanitize_hwaddress uwtable
define internal void @_ZN3mod6second18forward_of_reverse1g17heccf9262a2b6db9cE(ptr align 4 %0, ptr align 4 %1) unnamed_addr #14 !dbg !11715 {
  %3 = alloca ptr, align 8
  %4 = alloca ptr, align 8
  %5 = alloca [2 x float], align 4
  %6 = alloca [2 x float], align 4
  store ptr %0, ptr %4, align 8
  call void @llvm.dbg.declare(metadata ptr %4, metadata !11719, metadata !DIExpression()), !dbg !11723
  store ptr %1, ptr %3, align 8
  call void @llvm.dbg.declare(metadata ptr %3, metadata !11720, metadata !DIExpression()), !dbg !11724
  call void @llvm.dbg.declare(metadata ptr %6, metadata !11721, metadata !DIExpression()), !dbg !11725
  %7 = getelementptr inbounds [2 x float], ptr %6, i64 0, i64 0, !dbg !11726
  store float 0.000000e+00, ptr %7, align 4, !dbg !11726
  %8 = getelementptr inbounds [2 x float], ptr %6, i64 0, i64 1, !dbg !11726
  store float 0.000000e+00, ptr %8, align 4, !dbg !11726
  %9 = call float @_ZN3mod6second18forward_of_reverse2df17hca2877514b75e09cE(ptr align 4 %0, ptr align 4 %6, float 1.000000e+00), !dbg !11727
  call void @llvm.memcpy.p0.p0.i64(ptr align 4 "enzyme_type"="{[-1]:Pointer, [-1,0]:Float@float, [-1,4]:Float@float}" %5, ptr align 4 "enzyme_type"="{[-1]:Pointer, [-1,0]:Float@float, [-1,4]:Float@float}" %6, i64 8, i1 false), !dbg !11728
  call void @llvm.memcpy.p0.p0.i64(ptr align 4 "enzyme_type"="{[-1]:Pointer, [-1,0]:Float@float, [-1,4]:Float@float}" %1, ptr align 4 "enzyme_type"="{[-1]:Pointer, [-1,0]:Float@float, [-1,4]:Float@float}" %5, i64 8, i1 false), !dbg !11729
  ret void, !dbg !11730
}
)
num_fnc_args: 2
input_activity.len(): 2
rustc: /home/jed/src/rust-enzyme/build/x86_64-unknown-linux-gnu/llvm/include/llvm/IR/Instructions.h:3144: llvm::Value* llvm::ReturnInst::getOperand(unsigned int) const: Assertion `i_nocapture < OperandTraits<ReturnInst>::operands(this) && "getOperand() out of range!"' failed.
warning: `samples` (test "mod") generated 2 warnings
error: could not compile `samples` (test "mod"); 2 warnings emitted
```